WebsiteDataStore::setTrackingPreventionEnabled should not resolve directories

szewai opened the pull request:
#### 15365541a8138ba8f48a423fec7fc52acd5ce2fe
<pre>
WebsiteDataStore::setTrackingPreventionEnabled should not resolve directories
<a href="https://bugs.webkit.org/show_bug.cgi?id=270148">https://bugs.webkit.org/show_bug.cgi?id=270148</a>
<a href="https://rdar.apple.com/123669767">rdar://123669767</a>

Reviewed by Brent Fulgham.

When investigating a performance issue, I noticed WebsiteDataStore::setTrackingPreventionEnabled spent a long time on
file operations. It turns out setTrackingPreventionEnabled tries to resolve directories, which is unnecessary for this
operation -- it does not need directory information at all.

* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::setTrackingPreventionEnabled):

Canonical link: <a href="https://commits.webkit.org/275404@main">https://commits.webkit.org/275404@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bc36cddf9cfb43cdc8d6ada24b330e1d0e731b81

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20689 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44056 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44246 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37768 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23809 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18019 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34437 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42249 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17607 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35885 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15104 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15303 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45628 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37851 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37226 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40970 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16490 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39393 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18109 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9358 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18165 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17753 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
```diff
@@ -1672,24 +1672,14 @@ void WebsiteDataStore::setTrackingPreventionEnabled(bool enabled)
-    if (enabled) {
-        m_trackingPreventionEnabled = true;
-        
-        resolveDirectoriesIfNecessary();
```
Are we sure that directory resolution is always done before we need to access a directory? It seems correct that we should not do this work in this method, but I want to make sure we resolve directories before we need to access them.

brentfulgham replied:
It looks like we always do this when initializing a new WebProcess, so it seems like this should always be in the correct state, even without this call.

szewai replied:
`resolveDirectoriesIfNecessary` is called when we get the parameters of the WebsiteDataStore, and that includes when we create web process and when we add session to network process.

brentfulgham reviewed (approved):
Looks good, r=me!

szewai committed:
WebsiteDataStore::setTrackingPreventionEnabled should not resolve directories
https://bugs.webkit.org/show_bug.cgi?id=270148
rdar://123669767

Reviewed by Brent Fulgham.

When investigating a performance issue, I noticed WebsiteDataStore::setTrackingPreventionEnabled spent a long time on
file operations. It turns out setTrackingPreventionEnabled tries to resolve directories, which is unnecessary for this
operation -- it does not need directory information at all.

* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::setTrackingPreventionEnabled):

Canonical link: https://commits.webkit.org/275404@main
